src/stores/Binding/slices/FieldSelection.ts

```diff
@@ -74,30 +84,61 @@ export const getStoreWithFieldSelectionSettings = (
         );
     },
 
-    setSingleSelection: (bindingUUID, field, selectionType) => {
+    setSearchQuery: (value) => {
+        set(
+            produce((state: BindingState) => {
+                state.searchQuery = value;
+            }),
+            false,
+            'Search Query Set'
+        );
+    },
+
+    setSelectionSaving: (value) => {
+        set(
+            produce((state: BindingState) => {
+                state.selectionSaving = value;
+            }),
+            false,
+            'Selection Saving Set'
+        );
+    },
+
+    setMultiSelection: (bindingUUID, updatedFields) => {
+        const { selections } = get();
+
+        const fields = selections[bindingUUID];
```

Comment:
This should be removed and put into the set itself. That way we reduce the chance of issues with `get` and `produce` being used together causing issues.